include/fiber_tasking_lib/task_scheduler.h

```diff
@@ -181,19 +183,21 @@ class TaskScheduler {
 	/**
 	 * Yields execution to another task until counter == value
 	 *
-	 * @param counter    The counter to check
-	 * @param value      The value to wait for
+	 * @param counter             The counter to check
+	 * @param value               The value to wait for
+	 * @param pinToCurrentThread  If true, the task invoking this call will not resume on a different thread
 	 */
-	void WaitForCounter(std::shared_ptr<std::atomic_uint> &counter, uint value);
+	void WaitForCounter(std::shared_ptr<std::atomic_uint> &counter, uint value, bool pinToCurrentThread = false);
 
-private:
 	/**
 	 * Gets the 0-based index of the current thread
 	 * This is useful for m_tls[GetCurrentThreadIndex()]
 	 *
 	 * @return    The index of the current thread
 	 */
 	std::size_t GetCurrentThreadIndex();
+
+private:
```

Comment:
Oops, that is my bad. I exposed this function unrelated to this change, I think it is useful to make it public. When you are running a task, you can use the thread index to select the data you are working on, for example if you have an array of values to process, you can process modulo thread index elements to forgo synchronisation. If you feel this is not appropriate, I'll remove it.